python/lsst/meas/algorithms/detection.py

```diff
@@ -799,20 +799,17 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True,
 
         return results
 
-    def removeBadPixels(self, middle, mask):
+    def removeBadPixels(self, middle):
         """Set the significance of flagged pixels to zero.
 
         Parameters
         ----------
-        middle : `lsst.afw.image.Exposure` or `lsst.afw.image.MaskedImage`
-            Convolved image to zero certain mask planes from.
+        middle : `lsst.afw.image.Exposure`
+            Score or maximum likelihood difference image.
             The image plane will be modified in place.
-        mask : `lsst.afw.image.Mask`
-            Mask to select planes from to zero the image; will be restricted
-            to the image's bounding box.
         """
-        badPixelMask = mask.getPlaneBitMask(self.config.excludeMaskPlanes)
-        badPixels = mask.subset(middle.getBBox()).array & badPixelMask > 0
+        badPixelMask = lsst.afw.image.Mask.getPlaneBitMask(self.config.excludeMaskPlanes)
```

Comment:
Don't set this back to what it was, since the global definition is not necessarily correct. Use `middle.mask.getPlaneBitMask(..`